haystack/nodes/prompt/prompt_node.py

```diff
@@ -97,24 +93,16 @@ def __init__(
             },
         )
         super().__init__()
-        self.prompt_templates: Dict[str, PromptTemplate] = {pt.name: pt for pt in get_predefined_prompt_templates()}  # type: ignore
-        self.default_prompt_template: Union[str, PromptTemplate, None] = default_prompt_template
+        self._prompt_templates_cache: Dict[str, PromptTemplate] = {}
+        self._default_prompt_template = None
+        self.default_prompt_template = default_prompt_template
```

Comment:
We can omit the `self._default_prompt_template = None` right?

Reply:
We have to initialize the attribute and pylint complains if you do that outside of `__init__`.

Reply:
Let's add a comment to explain why we do it then.